x-pack/plugins/security_solution/cypress/integration/alerts_detection_callouts_readonly.spec.ts

```diff
@@ -0,0 +1,131 @@
+  const ALERTS_CALLOUT = 'read-only-access-to-alerts';
+  const RULES_CALLOUT = 'read-only-access-to-rules';
+
+  before(() => {
+    // First, we have to open the app on behalf of a priviledged user in order to initialize it.
+    // Otherwise the app will be disabled and show a "welcome"-like page.
+    loginAndWaitForPageWithoutDateRange(DETECTIONS_URL, ROLES.platform_engineer);
+    waitForAlertsIndexToBeCreated();
+
+    // After that we can login as a read-only user.
+    login(ROLES.reader);
+  });
+
+  after(() => {
+    deleteRoleAndUser(ROLES.reader);
+    deleteRoleAndUser(ROLES.platform_engineer);
+    removeSignalsIndex();
+  });
+
+  context('On Detections home page', () => {
+    beforeEach(() => {
+      loadPageAsReadOnlyUser(DETECTIONS_URL);
+    });
+
+    it('We show one primary callout', () => {
+      waitForCallOutToBeShown(ALERTS_CALLOUT, 'primary');
+    });
+
+    context('When a user clicks Dismiss on the callout', () => {
+      it('We hide it and persist the dismissal', () => {
+        waitForCallOutToBeShown(ALERTS_CALLOUT, 'primary');
+        dismissCallOut(ALERTS_CALLOUT);
+        reloadPage();
+        getCallOut(ALERTS_CALLOUT).should('not.exist');
+      });
+    });
+  });
+
+  context('On Rules Management page', () => {
+    beforeEach(() => {
+      loadPageAsReadOnlyUser(DETECTIONS_RULE_MANAGEMENT_URL);
+    });
+
+    it('We show one primary callout', () => {
+      waitForCallOutToBeShown(RULES_CALLOUT, 'primary');
+    });
+
+    context('When a user clicks Dismiss on the callout', () => {
+      it('We hide it and persist the dismissal', () => {
+        waitForCallOutToBeShown(RULES_CALLOUT, 'primary');
+        dismissCallOut(RULES_CALLOUT);
+        reloadPage();
+        getCallOut(RULES_CALLOUT).should('not.exist');
+      });
+    });
+  });
+
+  context('On Rule Details page', () => {
+    beforeEach(() => {
+      createCustomRule(newRule);
+
+      loadPageAsReadOnlyUser(DETECTIONS_RULE_MANAGEMENT_URL);
+      waitForPageTitleToBeShown();
+      goToRuleDetails();
+    });
+
+    afterEach(() => {
+      deleteCustomRule();
+    });
+
+    it('We show two primary callouts', () => {
+      waitForCallOutToBeShown(ALERTS_CALLOUT, 'primary');
+      waitForCallOutToBeShown(RULES_CALLOUT, 'primary');
+    });
+
+    context('When a user clicks Dismiss on the callouts', () => {
+      it('We hide them and persist the dismissal', () => {
+        waitForCallOutToBeShown(ALERTS_CALLOUT, 'primary');
+        waitForCallOutToBeShown(RULES_CALLOUT, 'primary');
+
+        dismissCallOut(ALERTS_CALLOUT);
+        reloadPage();
+
+        getCallOut(ALERTS_CALLOUT).should('not.exist');
+        getCallOut(RULES_CALLOUT).should('be.visible');
+
+        dismissCallOut(RULES_CALLOUT);
+        reloadPage();
+
+        getCallOut(ALERTS_CALLOUT).should('not.exist');
+        getCallOut(RULES_CALLOUT).should('not.exist');
+      });
+    });
+  });
+});
+
+const loadPageAsReadOnlyUser = (url: string) => {
```

Comment:
Can we move all these methods to the tasks folder?

Reply:
Hmm 🤔 , to be honest I'm not sure!

I moved functions that seem to me truly reusable, like some actions on callouts.

This code, on the other hand, is quite coupled to the test suite:

```
const loadPageAsReadOnlyUser = (url: string) => {
  waitForPageWithoutDateRange(url, ROLES.reader);
  waitForPageTitleToBeShown();
};

const reloadPage = () => {
  cy.reload();
  waitForPageTitleToBeShown();
};

const waitForPageTitleToBeShown = () => {
  cy.get(PAGE_TITLE).should('be.visible');
};
```

Here, for example, it waits for the page title. This looks natural for this particular suite only because callouts are rendered on top of the page title. Other suites might have different needs in terms of what to wait for when we `reloadPage()` - they might need to wait less or more to pass and also be resilient.

Please let me know if you have any strong opinion on that!

Reply:
I completely understand your point :) 

We are used to work in the other way, so, if for whatever reason another colleague wants to extend the tests, will probably create a new spec in order to don't overload the current one. And most likely is going to check the tasks folder for the methods he can use rather than check the spec for them.

If you think that the scenario I mentioned is not going to happen and you want to keep it on the spec file I would suggest you to move it at the top of the file after the imports to give a bit of visibility ;)

Reply:
Sure thing, easy! Did it. Thank you @MadameSheema !